app/src/main/java/com/vci/vectorcamapp/core/data/upload/image/ImageUploadWorker.kt

```diff
@@ -2,85 +2,491 @@ package com.vci.vectorcamapp.core.data.upload.image
+                        currentChunkSize =
+                            (currentChunkSize * SUCCESS_CHUNK_SIZE_MULTIPLIER).coerceAtMost(
+                                MAX_CHUNK_SIZE_BYTES
+                            )
+                        Log.i(
+                            "ImageUploadWorker",
+                            "Increasing chunk size to $currentChunkSize bytes."
+                        )
+                        successfulUploadsInARow = 0
+                    }
+                }
+
+                is DomainResult.Error -> {
+                    val error = chunkResult.error
+                    successfulUploadsInARow = 0
+                    currentChunkSize =
+                        (currentChunkSize / FAILURE_CHUNK_SIZE_DIVIDER).coerceAtLeast(
+                            MIN_CHUNK_SIZE_BYTES
+                        )
+
+                    if (error == NetworkError.TUS_PERMANENT_ERROR) {
+                        return DomainResult.Error(error)
+                    }
+
+                    recoveryAttempts++
+                    if (recoveryAttempts >= MAX_RETRIES) {
+                        Log.e(
+                            "ImageUploadWorker",
+                            "Exceeded max recovery attempts for a single image upload. Failing."
+                        )
+                        return DomainResult.Error(error)
+                    }
+
+                    try {
+                        uploader = tusClient.resumeOrCreateUpload(upload)
+                    } catch (resumeException: Exception) {
+                        if (isStopped) {
+                            throw CancellationException("Worker was stopped by the system.", resumeException)
+                        }
+                        Log.e("ImageUploadWorker", "Failed to resume upload.", resumeException)
+                        return DomainResult.Error(NetworkError.UNKNOWN_ERROR)
+                    }
+                }
+            }
+        }
+        Log.d("ImageUploadWorker", "Upload loop finished. Final offset: ${uploader.offset}")
+        return DomainResult.Success(Unit)
+    }
+
+    private fun createTusUpload(
+        file: File,
+        fingerprint: String,
+        metadata: Map<String, String>
+    ): TusUpload {
+        return TusUpload(file).apply {
+            this.fingerprint = fingerprint
+            this.metadata = metadata
+        }
     }
 
+    private fun calculateMD5(file: File): String {
+        val md = MessageDigest.getInstance("MD5")
+        FileInputStream(file).use { fis ->
+            val buffer = ByteArray(BYTE_ARRAY_SIZE)
+            var read: Int
+            while (fis.read(buffer).also { read = it } != -1) {
+                md.update(buffer, 0, read)
+            }
+        }
+        return md.digest().joinToString("") { "%02x".format(it) }
+    }
+
+    private suspend fun prepareFile(
+        resolver: ContentResolver,
```

Comment:
worker gives you access to context as a variable. No need to pass in contentResolver